### ISSUE TITLE
Uploads fail due to 'Upload not in PREPARING state' error

### ISSUE THREAD

**Issue by avigil-twist** [sbg/sevenbridges-python#80]
Occasionally when synchronously uploading files, the upload will fail with an error:
```
Exception in thread Thread-734:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/local/lib/python2.7/site-packages/sevenbridges/transfer/upload.py", line 491, in run
    'Can not start. Upload not in PREPARING state.'
SbgError: Can not start. Upload not in PREPARING state.
```

Testing with:
```
for i in range(500):
    upload = api.files.upload('test.txt', project='...', file_name='test%d.txt' % i, overwrite=True)
    assert upload.result(), i
```

The failed upload usually works on the second attempt. It also seems like the above exception happens in another thread, so the only indication of failure in the calling thread is an empty upload result. The above code block fails at the `assert` line:
```
Exception in thread Thread-145:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/local/lib/python2.7/site-packages/sevenbridges/transfer/upload.py", line 491, in run
    'Can not start. Upload not in PREPARING state.'
SbgError: Can not start. Upload not in PREPARING state.

---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-3-07c9ffd2dc0a> in <module>()
      3 for i in range(500):
      4     upload = api.files.upload('test.txt', project='...', file_name='test%d.txt' % i, overwrite=True)
----> 5     assert upload.result(), i
      6 
      7 

AssertionError: 12
```

I am currently working around this by retrying the upload when result is `None`. Tested using latest release 0.7.4.

**Comment by SenadI** [comment]
@avigil-twist Thank you for reporting this issue. I will fix in in the next version which should be coming soon.

**Comment by SenadI** [comment]
Fixed in 0.8.0
